src/midi/midi_mt32.cpp

```diff
@@ -683,6 +688,7 @@ void MidiHandler_mt32::Close()
 
 	// Stop the synthesizer
 	if (service) {
+		const std::lock_guard<std::mutex> lock(service_mutex);
 		service->closeSynth();
 		service->freeContext();
 	}
```

Comment:
I think it's good to be cautious—just because the issue hasn't manifested yet, it doesn't mean it never will if it can theoretically happen.

Nice explanatory commit message 👍🏻 There's one type at the end: because these